controllers/chat/ChatController.js

```diff
@@ -565,15 +565,15 @@ module.exports = {
         })
       );
 
-      const {betterChannelMember, betterChannelMemberObject} =
+      const {betterChannelMember, betterChannelMemberObject, updatedChannel} =
         await BetterSocialCore.chat.updateBetterChannelMembers(channel, findOrCreateChannel, true, {
           channel_type: CHANNEL_TYPE.ANONYMOUS
         });
 
       await client.disconnectUser();
 
       return res.status(200).json({
-        ...findOrCreateChannel,
+        ...updatedChannel,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change seems to be a good improvement as it introduces the `updatedChannel` variable which is more descriptive and clear about its purpose. However, I would suggest adding error handling around the `updateBetterChannelMembers` function call. If this function fails for any reason, it could lead to an unhandled promise rejection.

```diff
      const {betterChannelMember, betterChannelMemberObject, updatedChannel} =
         await BetterSocialCore.chat.updateBetterChannelMembers(channel, findOrCreateChannel, true, {
           channel_type: CHANNEL_TYPE.ANONYMOUS
         }).catch(error => {
           console.error('Error updating channel members:', error);
           return res.status(500).json({ error: 'Failed to update channel members' });
         });
 
       await client.disconnectUser();
```
This way, if there's an error during the execution of `updateBetterChannelMembers`, we log the error and send a response with status 500 indicating that something went wrong. This will prevent potential unhandled promise rejections and provide more informative error messages.


<!-- This is an auto-generated comment by OSS CodeRabbit -->